guides/developer/plugin-creators/overview.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: single
+title: "Overview"
+sidebar:
+  nav: guides
+---
+
+{% include toc %}
+
+<div class="notice--danger">
+  <strong>Note:</strong> Plugins are an alpha feature that is under active development and may change.
+</div>
+
+
+# Requirements
+
+* Spinnaker v1.19.0
+* Orca branch `release-1.19.x` (for local testing)
+* IntelliJ IDEA (for local testing)
+
+# Plugin overview
+
+A plugin enables an operator to extend Spinnaker with custom functionality. Use cases include fetching credentials from a custom authorization service, adding a wait stage to a pipeline, updating a Jira ticket, and sending Echo events to third-party tools.
+
+You can extend Spinnaker functionality with three types of plugins: ExtensionPoint, Interface (Non-ExtensionPoint), and Spring.
+
+# ExtensionPoint plugins
+
+Spinnaker uses the [Plugin Framework for Java (PF4J)](https://github.com/pf4j/pf4j) to indicate an _extension point_ interface to a service. You can create a plugin that implements the methods declared in an extension point.  Creating a plugin based on an extension point has a number of advantages:
+
+* It's the easiest - use the `@Extension` annotation and implement the methods declared in your chosen extension point
+* Spinnaker loads the plugin in an isolated classpath
+* It has the least amount of maintenance work; updates to Spinnaker are not likely to break your plugin
+
+## Finding an extension point
+
+An extension point is an interface that extends `org.pf4j.ExtensionPoint` and is located in the `api` module of a service. Spinnaker exposes the following extension points:
+
+* Orca
+  - [SimpleStage](OrcaSimpleStage) for creating a custom pipeline stage
+  - [PreconfiguredJobConfigurationProvider](OrcaPreconfiguredJobConfigurationProvider) for provisioning preconfigured Job stages
+
+* Echo
+  - [EventListener](EchoEventListener) for processing events posted into Echo
+
+Look through the code or ask in the Spinnaker Slack to find extension points not listed here.
+
+## Example ExtensionPoint plugin
+
+The [pf4jStagePlugin](https://github.com/spinnaker-plugin-examples/pf4jStagePlugin) creates a custom pipeline stage that waits a specified number of seconds before signaling success. Consult the [ExtensionPoint plugin walkthrough] for a detailed explanation of this plugin.
+
+# Interface (Non-ExtensionPoint) plugins
+
+The second way you can create a plugin is to implement a regular Java interface that you find in a service. Your plugin uses the PF4J `@Extension` annotation but does not extend a PF4J extension point.    
+
+Advantages:
+* Spinnaker loads the plugin in classpath isolation
```

Comment:
This is _slightly_ different from how you said it above. If there's no difference between the advantage here and the advange for extension point-style plugins, I'd say it exactly the same in both places.